### ISSUE TITLE
Unified PyPI-Conda dependency resolution

### ISSUE THREAD

**Issue by ruben-arts** [prefix-dev/pixi#2559]
While pixi can now avoid broken environments by figuring out what will be installed as a Conda package, then avoiding installing that package as a PyPI package. It doesn't solve the two ecosystems in one solve. Meaning it doesn't try to find new solutions when the Conda solve succeeded and then the PyPI one fails.

# Simple Scenario
- You depend on `numpy = "*"` as a conda dependency. 
- It solves for conda as `numpy ==2.1.3`
- You depend on a pypi-dependency that requires `numpy <2`
- Error: 
```
  × failed to solve the pypi requirements of 'default' 'linux-64'
  ├─▶ failed to resolve pypi dependencies
  ╰─▶ Because package_x==1.2.3 depends on numpy>=1.24.3,<2 and numpy==2.1.3, we can
      conclude that package_x==1.2.3 cannot be used.
      And because only package_x<=1.2.3 is available and you require
      package_x>=1.2.3, we can conclude that your requirements are unsatisfiable.
```

# Idea for a Solution
This would be solved if our Conda(Rattler) and PyPI(uv) solver implementations could interact and share the information about the problems they run into. Like if the example error shows up, the Conda solver could retry with that information and see if it is possible to get `numpy` with a lower version.

# Current workaround
You'll need to help the solvers to understand eachother needs. In the previous example you can tell the Conda dependencies to get `numpy = "<2"` it would have most likely solved your issues. 

# Disclaimer
This issue is for informational purpose. This is an extremely hard problem we're not planning to solve any time soon while we're working on other features. Feel free to discuss or share experiences to better map this issue.

**Comment by ruben-arts** [comment]
@abkfenris To verify you mean having the `pypi-dependencies` versions injected as constraints in the `conda` solve?
For example, in this `pixi.toml`:
```toml
[pypi-dependencies]
numpy = "<2"

[dependencies]
package_that_depends_on_numpy = "*"
```

It should result in a `numpy` lower then 2 in the conda environment?

**Comment by abkfenris** [comment]
Yes, that as well, but I was thinking of something a bit smaller in scope like this:

```toml
[pypi-dependencies]
numpy = "<2"

[dependencies]
numpy = "inherit-pypi-constraint"
```

Then the conda solve would use the constraint from `pypi-dependencies`.

I'm thinking it might be a step on the way to broader unified dependency resolution.
